### PR TITLE
broot: install man page and shell completions

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Man Page generation based on logic in [broot's release.sh](https://github.com/Canop/broot/blob/4d44b1585cde2b95e4742ca8615bd9c0b2772072/release.sh#L16-L18), which is run for GitHub release:
```sh
cp man/page build/broot.1
sed -i "s/#version/$version/g" build/broot.1
sed -i "s/#date/$(date +'%Y\/%m\/%d')/g" build/broot.1
```

Shell completions install based on ripgrep.

**Original Formula Output**
```console
❯ brew install broot
...
==> Pouring broot-1.2.7.arm64_big_sur.bottle.tar.gz
🍺  /opt/homebrew/Cellar/broot/1.2.7: 8 files, 5.5MB

❯ man broot
No manual entry for broot

❯ head -n5 /opt/homebrew/share/zsh/site-functions/_broot
head: /opt/homebrew/share/zsh/site-functions/_broot: No such file or directory
```

**New Formula Output**
```console
❯ brew install --build-from-source broot
...

❯ man broot
broot(1)                         broot manpage                        broot(1)

...

1.2.7                             2021/03/07                          broot(1)


❯ head -n5 /opt/homebrew/share/zsh/site-functions/_broot
#compdef broot

autoload -U is-at-least

_broot() {
```

**New Formula HEAD man page**
```console
❯ brew install --build-from-source broot --head
...

❯ man broot
broot(1)                         broot manpage                        broot(1)

...

HEAD-4d44b15                      2021/03/07                          broot(1)
```

### CODE REVIEW

File: Formula/broot.rb [hunk 1]
```diff
@@ -19,6 +20,24 @@ class Broot < Formula
 
   def install
     system "cargo", "install", *std_cargo_args
+
+    # Replace man page "#version" and "#date" based on logic in release.sh
+    build_date = Utils.safe_popen_read("date -u +'%Y/%m/%d' 2> /dev/null").chomp
+    inreplace "man/page" do |s|
+      s.gsub! "#version", version
+      s.gsub! "#date", build_date
+    end
```
Comment (SMillerDev) [inline Formula/broot.rb:29]: We're downloading a release. Shouldn't this be done already?

File: Formula/broot.rb [hunk 2]
```diff
@@ -19,6 +20,24 @@ class Broot < Formula
 
   def install
     system "cargo", "install", *std_cargo_args
+
+    # Replace man page "#version" and "#date" based on logic in release.sh
+    build_date = Utils.safe_popen_read("date -u +'%Y/%m/%d' 2> /dev/null").chomp
+    inreplace "man/page" do |s|
+      s.gsub! "#version", version
+      s.gsub! "#date", build_date
```
Comment (carlocab) [inline Formula/broot.rb:28]:
```suggestion
    inreplace "man/page" do |s|
      s.gsub! "#version", version
      s.gsub! "#date", Time.now.strftime("%Y/%m/%d")
```
We don't need to shell out to `date` here.

File: Formula/broot.rb (hunk 1 above)
Comment (cho-m) [reply Formula/broot.rb:29]: It looks like the man page is only added into pre-built GitHub zip archive.
I think Homebrew uses source code for release (`broot-<version>.tar.gz`), which only has the unmodified `./man/page` file.

File: Formula/broot.rb (hunk 2 above)
Comment (cho-m) [reply Formula/broot.rb:28]: Sounds good. I was mainly basing on how some other formulas were doing it. I'll try modifying and rerunning.

Is there a preference of running using local time or UTC (`Time.now.utc.strftime("%Y/%m/%d")`)?

Comment (carlocab) [reply Formula/broot.rb:28]: Not really; not sure it matters that much. Use whichever you prefer.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**